### PR TITLE
SONARIAC-1067 Update Sonar-Plugin-API to Version 10.0.0.695

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -198,6 +199,12 @@
         <version>${version.archunit}</version>
         <scope>test</scope>
       </dependency>
+      <dependency>
+        <groupId>org.slf4j</groupId>
+        <artifactId>slf4j-api</artifactId>
```
Comment (petertrr) [inline pom.xml:204]: Do we need to include `slf4j-simple` for tests anywhere? I can see in logs on CI that for `sonar-iac-plugin`
```
[INFO] --- surefire:2.22.2:test (default-test) @ sonar-iac-plugin ---
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
```

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesSensorTest.java [hunk 2]
```diff
@@ -104,7 +105,17 @@ void shouldReturnKubernetesDescriptor() {
   @Test
   void shouldNotParseYamlFileWithHelmTemplateDirectives() {
     analyse(sensor(), inputFile(K8_IDENTIFIERS + "{{ .Values.count }}"));
-    asserNotSourceFileIsParsed();
+    assertNotSourceFileIsParsed();
+  }
+
+  @Test
+  void shouldParseYamlFileWithIdentasdifiers() throws IOException {
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesSensorTest.java:112]:
```suggestion
  void shouldParseYamlFileWithIdentifiers() throws IOException {
```

**Review by petertrr — COMMENTED** [review]
Just my two cents; I'm not that familiar with what the update brings

File: pom.xml [hunk 3]
```diff
@@ -206,6 +213,12 @@
         <version>${version.sonar.api}</version>
         <scope>provided</scope>
       </dependency>
+      <dependency>
+        <groupId>org.sonarsource.api.plugin</groupId>
+        <artifactId>sonar-plugin-api-test-fixtures</artifactId>
+        <version>${version.sonar.api}</version>
+        <scope>test</scope>
```
Comment (jonas-wielage-sonarsource) [inline pom.xml:220]: I realized this should be under test-dependencies, not provided dependencies

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! I've added some questions but they are not blocking the PR.

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesSensorTest.java (hunk 2 above)
Comment (nils-werner-sonarsource) [reply iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesSensorTest.java:112]: Nice catch

File: iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java [hunk 4]
```diff
@@ -228,7 +228,8 @@ private void logParsingError(ParseException e) {
       if (detailedMessage != null) {
         LOG.debug(detailedMessage);
       }
-      LOG.debug(getStackTrace(e));
+      String stackTrace = getStackTrace(e);
+      LOG.debug(stackTrace);
```
[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java:232]: Code smell from sonar...

File: iac-common/src/main/java/org/sonar/iac/common/extension/DurationStatistics.java [hunk 5]
```diff
@@ -94,7 +94,8 @@ public void log() {
           .append(" ")
           .append(format.format(e.getValue().get() / 1_000_000L))
           .append(" ms"));
-      LOG.info(out.toString());
+      String outString = out.toString();
+      LOG.info(outString);
```
[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/extension/DurationStatistics.java:98]: Code smell from sonar...

[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/extension/DurationStatistics.java:98]: This would produce a code smell.

File: pom.xml [hunk 6]
```diff
@@ -215,7 +228,10 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
-          <configuration />
+          <configuration>
+            <redirectTestOutputToFile>true</redirectTestOutputToFile>
+            <reportsDirectory>${project.build.directory}/test-reports</reportsDirectory>
+          </configuration>
```
Comment (jonas-wielage-sonarsource) [inline pom.xml:234]: @petertrr WDYT?

Comment (petertrr) [reply pom.xml:234]: I wonder if we get any output from failing test or only a name. Anyway, both would be sufficient as it is usually easy to reproduce test failure locally

Comment (jonas-wielage-sonarsource) [reply pom.xml:234]: Yes, we get the output if a test fails.

**Pull request merged**